### PR TITLE
Introduce KeyVaultCredential for SDK v2

### PR DESCRIPTION
**Pull request by lodrem**
#### What type of PR is this?

/feature

#### What this PR does / why we need it:

This PR introduces KeyVaultCredential, which fetches tokens from the KeyVault secret. These tokens are used for accessing resources from NetworkTenant or as auxiliary tokens to access resources from MultiTenant.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/azclient/armauth/keyvault_credential.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+	token *azcore.AccessToken
+}
+
+type KeyVaultCredentialSecret struct {
+	AccessToken string    `json:"access_token"`
+	ExpiresOn   time.Time `json:"expires_on"`
+}
+
+var secretURIRE = regexp.MustCompile("^(https://[a-zA-Z0-9-_]+\\.vault\\.azure\\.net)(/secrets/)([a-zA-Z0-9-_]+)/?$")
```
Comment (MartinForReal) [inline pkg/azclient/armauth/keyvault_credential.go:43]: this may not work in sovereign cloud.

Comment (lodrem) [reply pkg/azclient/armauth/keyvault_credential.go]: Good catch. Fixed and added test for `https://{vault-name}.vault.azure.cn`.

File: pkg/azclient/armauth/keyvault_credential.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+	token *azcore.AccessToken
+}
+
+type KeyVaultCredentialSecret struct {
+	AccessToken string    `json:"access_token"`
+	ExpiresOn   time.Time `json:"expires_on"`
+}
+
+var secretURIRE = regexp.MustCompile("^(https://[a-zA-Z0-9-_]+\\.vault\\.azure\\.(net|cn))(/secrets/)([a-zA-Z0-9-_]+)/?$")
```
Comment (feiskyer) [inline pkg/azclient/armauth/keyvault_credential.go:43]: this may not work on some clouds. Could you only validate for the specific cloud?

Comment (lodrem) [reply pkg/azclient/armauth/keyvault_credential.go]: I am thinking about removing the RE validation. PTAL Thanks.

File: pkg/azclient/armauth/auxiliary_auth_policy.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	tokens := make([]string, 0, len(p.credentials))
+
+	for _, cred := range p.credentials {
+		token, err := cred.GetToken(context.TODO(), policy.TokenRequestOptions{
+			Scopes: []string{p.scope},
+		})
+		if err != nil {
+			return nil, err
+		}
+		tokens = append(tokens, fmt.Sprintf("Bearer %s", token.Token))
```
Comment (MartinForReal) [inline pkg/azclient/armauth/auxiliary_auth_policy.go:55]: I think the token format should be Bearer token1,token2 https://github.com/Azure/azure-sdk-for-go/blob/50b2bf819d7f22f1c749c89f39898deaed7a0c02/sdk/azcore/arm/runtime/policy_bearer_token.go#L111C23-L111C51

**Pull request merged**